core/cap-0063.md

```diff
@@ -0,0 +1,496 @@
+
+Additionally, this CAP introduces automatic restoration via `InvokeHostFunctionOp`, where any archived key present in the
+footprint is automatically restored. Initially in protocol 20, the state archival design was not solidified enough to
+enable automatic restoration, so an explicit restore operation was required. While this was not technically required in
+protocol 20, it was important for contract interfaces to reason properly about restoration such that when full state
+archival was introduce, it would not break preexisting deployments. Given that the full State Archival design has been
+mostly finalized in [CAP-0057](cap-0057.md), it is now appropriate to introduce automatic restoration.
+
+## Specification
+
+### XDR changes
+
+```
+--- a/Stellar-contract-config-setting.x
++++ b/Stellar-contract-config-setting.x
+@@ -26,28 +26,28 @@ struct ConfigSettingContractComputeV0
+ // Ledger access settings for contracts.
+ struct ConfigSettingContractLedgerCostV0
+ {
+-    // Maximum number of ledger entry read operations per ledger
+-    uint32 ledgerMaxReadLedgerEntries;
+-    // Maximum number of bytes that can be read per ledger
+-    uint32 ledgerMaxReadBytes;
++    // Maximum number of live Soroban entry read operations per ledger
++    uint32 ledgerMaxReadLiveSorobanEntries;
++    // Maximum number of live Soroban bytes that can be read per ledger
++    uint32 ledgerMaxLiveSorobanBytes;
+     // Maximum number of ledger entry write operations per ledger
+     uint32 ledgerMaxWriteLedgerEntries;
+     // Maximum number of bytes that can be written per ledger
+     uint32 ledgerMaxWriteBytes;
+ 
+-    // Maximum number of ledger entry read operations per transaction
+-    uint32 txMaxReadLedgerEntries;
+-    // Maximum number of bytes that can be read per transaction
+-    uint32 txMaxReadBytes;
++    // Maximum number of live Soroban ledger entry read operations per transaction
++    uint32 txMaxReadLiveSorobanEntries;
++    // Maximum number of live Soroban bytes that can be read per transaction
++    uint32 txMaxReadLiveSorobanBytes;
+     // Maximum number of ledger entry write operations per transaction
+     uint32 txMaxWriteLedgerEntries;
+     // Maximum number of bytes that can be written per transaction
+     uint32 txMaxWriteBytes;
+ 
+-    int64 feeReadLedgerEntry;  // Fee per ledger entry read
+-    int64 feeWriteLedgerEntry; // Fee per ledger entry write
++    int64 feeReadLiveSorobanEntry;  // Fee per live soroban ledger entry read
++    int64 feeWriteLedgerEntry;      // Fee per ledger entry write
+ 
+-    int64 feeRead1KB;  // Fee for reading 1KB
++    int64 feeReadLiveSoroban1KB;    // Fee for reading 1KB of live Soroban state
+ 
+     // The following parameters determine the write fee per 1KB.
+     // Write fee grows linearly until bucket list reaches this size
+@@ -60,6 +60,26 @@ struct ConfigSettingContractLedgerCostV0
+     uint32 bucketListWriteFeeGrowthFactor;
+ };
+ 
++struct ConfigSettingContractLedgerCostExtV1
```

Comment:
I don't have too strong opinion, but it might make sense to use V1 struct with all the fields instead of ext here. Or at least call it V0Ext to clarify the connection between these two.